src/core/net/srp_server.cpp

```diff
@@ -386,7 +386,7 @@ void Server::CommitSrpUpdate(Error                    aError,
                 Service *newService = existingHost->AddService(service->GetFullName());
 
                 VerifyOrExit(newService != nullptr, aError = kErrorNoBufs);
-                SuccessOrExit(aError = newService->CopyResourcesFrom(*service));
+                newService->TakeResourcesFrom(*service);
```

Comment:
@abtink  Will it be easier for us to support sub-types if we simply adds new services to the existing host? For example:
```c++
// pseudo code for moving service from aHost to existingHost
existingHost->Remove(existingService);
aHost.Remove(service);
existingHost->Add(service);
service->mTimeLastUpdate = TimerMilli::GetNow();
```
(We will need to refactor the while-loop as we are deleting elements during iteration

Reply:
@wgtdkp I am not sure if we can use the `Remove` and `Add` model you are suggesting (basically replacing the existing service with the new `Service` being committed).

I think it is possible and valid for subtypes to be added or removed in seperate (follow up) SRP update messages for an existing service. In such a case, the `service` we parse from the new `message` and want to commit may not have the full list of previously registered sub-types (just the new ones being added/removed), so we cannot replace the existing one with new one being committed and need to combine/merge them.

------

_Some more info on the data model I am considering for the sub-types_:

The `Service` is mainly a representation of a `SRV` and `TXT` records we get for a "service instance name". We can have multiple `PTR` records all pointing to the same "service instance name" (basically different sub-types names all associated with same `SRV`/`TXT` info or the same `Service` object). 

So the data model I am considering for subtypes is to have a list of sub-type label `HeapString`s in the `Service` object (I am leaning towards using a linked-list for this though also considered a fixed-size array which may be easier to implement but adds a hard limit on number of sub-types). We also need to track the "update time" for each sub-type seperately since they be registered in different messages and at different times.  If you have any suggestions/thoughts on this, please let me know. Thanks. 😃

Reply:
okay, it makes sense if we want to support more flexible sub-types updating :+1: